exist-core/src/main/java/org/exist/xquery/functions/fn/FunStrLength.java

```diff
@@ -38,64 +32,90 @@
+            )
+    };
+
+    public FunStrLength(final XQueryContext context, final FunctionSignature signature) {
+        super(context, signature);
+    }
+
+    @Override
+    protected Tuple2<Expression, Integer> strictCheckArgumentType(Expression argument,
+            @Nullable final SequenceType argType, final AnalyzeContextInfo argContextInfo, final int argPosition,
+            int returnType) {
+        if (getArgumentCount() == 1 && (argContextInfo.getFlags() & LocationStep.DOT_TEST) == LocationStep.DOT_TEST) {
+            /*
+                fn:string-length has different behaviour with regards the context item...
+                See https://www.biglist.com/lists/lists.mulberrytech.com/xsl-list/archives/201906/msg00021.html
+                See https://github.com/eXist-db/exist/issues/2798
+             */
+            return Tuple(argument, returnType);
+        } else {
+            return super.strictCheckArgumentType(argument, argType, argContextInfo, argPosition, returnType);
+        }
+    }
+
+    @Override
+    public Sequence eval(Sequence contextSequence, final Item contextItem) throws XPathException {
         if (context.getProfiler().isEnabled()) {
-            context.getProfiler().start(this);       
+            context.getProfiler().start(this);
             context.getProfiler().message(this, Profiler.DEPENDENCIES, "DEPENDENCIES", Dependency.getDependenciesName(this.getDependencies()));
-            if (contextSequence != null)
-                {context.getProfiler().message(this, Profiler.START_SEQUENCES, "CONTEXT SEQUENCE", contextSequence);}
-            if (contextItem != null)
-                {context.getProfiler().message(this, Profiler.START_SEQUENCES, "CONTEXT ITEM", contextItem.toSequence());}
-        }  
-        
-        
-		if(contextItem != null)
-			{contextSequence = contextItem.toSequence();}
-        
-        
-		if(getSignature().getArgumentCount() == 1)
-			{contextSequence = getArgument(0).eval(contextSequence);}
-		
-		if (contextSequence == null)
-			{throw new XPathException(this, ErrorCodes.XPDY0002, "Undefined context item");}
-		
-		final String strval = contextSequence.getStringValue();
-
-		final Sequence result = new IntegerValue(FunStringToCodepoints.getCodePointCount(strval));
-        
-        if (context.getProfiler().isEnabled()) 
-            {context.getProfiler().end(this, "", result);} 
-        
-        return result;            
-	}
+            if (contextSequence != null) {
+                context.getProfiler().message(this, Profiler.START_SEQUENCES, "CONTEXT SEQUENCE", contextSequence);
+            }
+            if (contextItem != null) {
+                context.getProfiler().message(this, Profiler.START_SEQUENCES, "CONTEXT ITEM", contextItem.toSequence());
+            }
+        }
+
+
+        if (contextItem != null) {
+            contextSequence = contextItem.toSequence();
+        }
+
```

Comment:
surplus blank line

Reply:
As commented above.

Reply:
Well, if you ask me I would auto format any file in a pre-commit hook and would never have to care about these. It is very tedious.